### PR TITLE
assert_worker_story

### PR DESCRIPTION
**Pull request by crusaderky**
New tool to test the output of Worker.story
I'd like to have some opinions on the design before I continue with unit tests

### CODE REVIEW

File: distributed/utils_test.py [hunk 1]
```diff
@@ -1761,3 +1761,65 @@ async def connect(self, *args, **kwargs):
         return LockedComm(
             comm, self.read_event, self.read_queue, self.write_event, self.write_queue
         )
+
+
+def assert_story(story: list[tuple], expect: list[tuple]) -> None:
```
Comment (fjetter) [inline distributed/utils_test.py:1766]: Should this be called `assert_worker_story` since the scheduler still uses a different format?

**Review by fjetter — APPROVED** [review]
Overall looks good and very useful

File: distributed/utils_test.py (hunk 1 above)
Comment (crusaderky) [reply distributed/utils_test.py]: changed it

**Comment by crusaderky** [comment]
This is ready for final review and merge

File: distributed/utils_test.py [hunk 2]
```diff
@@ -1761,3 +1761,54 @@ async def connect(self, *args, **kwargs):
...
+
+    Parameters
+    ==========
+    story: list[tuple]
+        Output of Worker.story
+    expect: list[tuple]
+        Expected events. Each expected event must contain exactly 2 less fields than the
+        story (the last two fields are always the stimulus_id and the timestamp).
+
+    story may contain more events than expect. Extra events are ignored.
```
Comment (fjetter) [inline distributed/utils_test.py:1777]: I'm wondering if this is too relaxed. Consider

```python
[
    "ensure-exists",
    ("released", "waiting"),
    ("waiting", "executing"),
    ("executing", "memory"),
]
```

is quite different than

```python
[
    "ensure-exists",
    ("released", "waiting"),
    ("waiting", "executing"),
    ("executing", "cancelled"),
    ("cancelled", "released"),),
    ("released", "waiting"),
   ...
    ("executing", "memory"),
]
```

Comment (fjetter) [reply distributed/utils_test.py:1777]: I would assume we need both the relaxed and strict version occasionally?

Comment (crusaderky) [reply distributed/utils_test.py]: Added strict mode, please review

**Pull request merged**